central/policycategory/datastore/datastore_impl.go

```diff
@@ -32,11 +35,81 @@ var (
 )
 
 type datastoreImpl struct {
-	storage  store.Store
-	indexer  index.Indexer
-	searcher search.Searcher
+	storage              store.Store
+	indexer              index.Indexer
+	searcher             search.Searcher
+	policyCategoryEdgeDS datastore.DataStore
+	categoryMutex        sync.Mutex
+}
+
+func (ds *datastoreImpl) SetPolicyCategoriesForPolicy(ctx context.Context, policyID string, categoryNames []string) error {
+	ds.categoryMutex.Lock()
```

Comment:
check for ` !features.NewPolicyCategories.Enabled() || !features.PostgresDatastore.Enabled()` and throw unsupported. I didn't realize earlier that this datastore also exists on rocksdb.

Reply:
Ok I've blocked the calls to this func under feature flag check ^